packages/keychain/src/components/ConfirmTransaction.tsx

```diff
@@ -23,14 +25,40 @@ export function ConfirmTransaction() {
     });
   };
 
-  const policies = useMemo<Policy[]>(
+  const transactions = useMemo<Policy[]>(
     () =>
       (Array.isArray(ctx.transactions)
         ? ctx.transactions
         : [ctx.transactions]
-      ).map((c) => ({ target: c.contractAddress, method: c.entrypoint })),
+      ).map((c) => ({
+        target: c.contractAddress,
+        method: c.entrypoint,
+      })),
     [ctx.transactions],
   );
+  const updateSession = useMemo(() => {
+    if (policiesUpdated) return false;
+
+    const txnsApproved = transactions.every((transaction) =>
+      policies.some(
+        (policy) =>
+          addAddressPadding(policy.target) ===
```

Comment:
oh i didn't know about this util func. i should use that for formatAddress func.